proposals/text/3665-multi-task-component.md

```diff
@@ -0,0 +1,332 @@
+  used to pass data between nodes.
+
+  We can also use MultiTaskPipeline to achieve the same goal:
+
+  ``` python
+      top_k = 3
+      query = "Who are the parents of Arya Stark?"
+      qg = MultiTaskComponent(current_task="question-generation")
+      qa = MultiTaskComponent(current_task="question-answering")
+
+      pipe = MultiTaskPipeline(multi_tasks=[question_generation, question_answering], retriever=retriever)
+      result = pipe.run(query=query, questions={"bind_result": "MultiTaskComponent_0"})
+
+      print(result["results"])
+  ```
+
+  In the example above we show another way to bind the results of one MultiTaskComponent to the input of another by
+  using keyword arguments. The `questions` keyword parameter is used to bind the results of the `MultiTaskComponent_0`
+  node and will be available to all downstream nodes via `invocation context` (hidden inside MultiTaskPipeline).
+
+
+  - Default tasks/prompts to be added to MultiTaskComponent
+
+    [Muffin]:
+      - Summarization
+      - Natural Language Inference
+      - Multiple-Choice QA
+      - Translation
+      - Sentiment Analysis
+      - Extractive QA
+      - Structured Data to Text
+      - Coreference Resolution
+      - Code Repair
+      - Code Error Generation
+      - Dialogue Context Generation
+      - Closed-Book QA
+      - Next Sentence Prediction
+      - Paraphrasing Identification
+      - Conversational Question Answering
+      - Topic Classification
+      - Mathematical QA
+      - Dialog Next Turn Prediction
+      - Grammatical Acceptability
+      - Punctuation fixing
+
+    [T0-SF]:
+      - Adversarial QA
+      - Question Generation
+      - Commonsense Reasoning
+      - Title Generation
+      - Dialogue Turn Prediction
+      - Predict Span Indices
+      - Context Generation
+
+    [NIV2]:
+      - Program Execution
+      - Text Matching
+      - Toxic Language Detection
+      - Cause Effect Classification
+      - Information Extraction
+      - Textual Entailment
+      - Wrong Candidate Generation
+      - Named Entity Recognition
+      - Commonsense Classification
+      - Fill-in-the-blank
+      - Text Completion
+      - Sentence Composition
+      - Question Understanding
+
+    [CoT Reasoning]:
+      - Explanation Generation
+      - Generate Question And Answer
+      - Grade School Math Word Problems
+      - Algebraic Question Answering
+      - Common Sense Reasoning Over Entities
+      - Common Sense Reasoning For QA
+      - Passage Based Question Answering
+      - Sense-Making And Explanation
+
+  # Drawbacks
+- One potential drawback of MultiTaskComponent is that it may require a significant amount of computational resources
+  to use. This may limit its use in applications or environments where there are constraints on the available hardware
+  or software resources. In addition, MultiTaskComponent may not be suitable for all NLP tasks or use cases, and users
```

Comment:
The sentence beginning with "In addition ..." sounds like a separate point from the previous sentence so I would make it a separate bullet point.
```suggestion
  or software resources. 
  - In addition, MultiTaskComponent may not be suitable for all NLP tasks or use cases, and users
```